src/components/MultiSelect/MultiSelect.tsx

```diff
@@ -1,138 +1,433 @@
+        }
+
+        const filteredItems = items.filter((item) => {
+          if (item.children) {
+            if (item.name.toLowerCase().includes(value)) {
+              return true;
+            }
+
+            return item.children.some((childItem) =>
+              childItem.name.toLowerCase().includes(value)
+            );
+          }
+
+          return item.name.toLowerCase().includes(value);
+        });
+
+        setItemsList(filteredItems);
+      };
+
+      const clearSearchKeyword = () => {
+        if (listOverflow === "scroll") {
+          setItemsList(items);
+        } else {
+          listItemsCount === defaultItemsVisible
+            ? displayDefaultItems()
+            : setItemsList(items);
+        }
+      };
+
+      const displayDefaultItems = () => {
+        const list = [];
+        let count = 0;
+
+        for (let i = 0; i < items.length && count < defaultItemsVisible; i++) {
+          const currentItem = { ...items[i] };
+          count++;
+          if (currentItem.children) {
+            const showChilds = isAllChecked(id, currentItem);
+            const showIndeterminateChilds = isIndeterminate(id, currentItem);
+            count +=
+              !showChilds || !showIndeterminateChilds
+                ? currentItem.children?.length
+                : items[i].children.length;
+          }
+
+          list.push(currentItem);
+          if (count >= defaultItemsVisible) {
+            break;
+          }
+        }
+        setItemsList(list);
+      };
+
+      const showSearchInputBox = () => {
+        if (isSearchable) {
+          return (
+            <TextInput
+              id="multi-select-text-input-id"
+              labelText={`Search ${buttonText}`}
+              isClearable={true}
+              isClearableCallback={clearSearchKeyword}
+              isRequired
+              placeholder={`Search ${buttonText}`}
+              onChange={onChangeSearch}
+              showLabel={false}
+              showRequiredLabel={false}
+              step={1}
+              textInputType="default"
+              type="text"
+              __css={styles.menuSearchInputBox}
+              marginBottom="s"
+            />
+          );
+        }
+      };
+
+      const displayAccordionData = () => {
         return (
-          <MultiSelectListBox
-            {...commonProps}
-            ref={ref}
-            onChange={listboxOnChange}
-            {...rest}
-          />
+          <>
+            <Box>
+              {showSearchInputBox()}
+              {itemsNotFound()}
+              <CheckboxGroup
+                id="multi-select-checkbox-group"
+                layout="column"
+                isFullWidth
+                isRequired={false}
+                labelText=""
+                name="multi-select-checkbox-group"
+              >
+                {itemsList.map((item: MultiSelectItem) => (
+                  <React.Fragment key={item.id}>
```

Comment:
This is why you are getting the warning message: "NYPL Reservoir CheckboxGroup: Only `Checkbox` components are allowed as children."

The `React.Fragment` is a component when you add the key, so CheckboxGroup thinks you're passing something that is not a `Checkbox`.

What you can do is pass an array as a child to render to any React component, such as `[<Checkbox key={1} />, <Checkbox key={2} />, <Checkbox key={3} />, ...etc]` and this should work. This means you may need to rework this as a function to return the correct array and _not_ use `React.Fragment.